src/Tec/Plugin.php

```diff
@@ -104,26 +139,1020 @@ public function register() {
+		if ( ! empty ( $record[ $record_meta_key ] ) ) {
+
+			// If there are multiple connections, e.g. more organizers for an event.
+			if ( $connection['multiple'] ) {
+				$multiple = false;
+				$ids      = $this->maybe_explode( $record[ $record_meta_key ] );
+				foreach ( $ids as $id ) {
+					$this->old_linked_post_id = $id;
+					$record[ $record_meta_key ] = $id;
+					$update_successful = $this->update_linked_post_meta(
+						$connection['linked_post_type'],
+						! empty ( $connection['connection_meta_key'] ) ? $connection['connection_meta_key'] : $record_meta_key,
+						$post_id,
+						$record,
+						$multiple
+					);
+
+					$msg = $multiple ? "Adding " : "Updating ";
+					$msg .= "metadata `" . $record_meta_key . "` for " . $post_title . " was ";
+					$msg .= $update_successful ? "successful" : "NOT successful (or linked post doesn't exist)";
+					$this->add_to_log( $msg );
+					// Set to `true` after first.
+					$multiple = true;
+
+					// Update the ticket IDs in the metadata
+					if ( $post_type == 'tec_tc_order' && $record_meta_key == '_tec_tc_order_tickets_in_order' && $update_successful ) {
+						$this->replace_ids_in_metavalue( $post_id );
+					}
+				}
+			} else {
+				$update_successful = $this->update_linked_post_meta( $connection['linked_post_type'], $record_meta_key, $post_id, $record );
+				$msg = "Updating metadata `" . $record_meta_key . "` for " . $post_title . " was ";
+				$msg .= $update_successful ? "successful" : "NOT successful";
+				$this->add_to_log( $msg );
+			}
+		}
+	}
+
+	/**
+	 * Re-save _tribe_default_ticket_provider for tribe_event.
+	 * WP All Import runs wp_unslash() and destroys "TEC\Tickets\Commerce\Module"
+	 *
+	 * @param array $record  The post data.
+	 * @param int   $post_id The new post ID.
+	 *
+	 * @return void
+	 */
+	public function resave_ticket_provider_for_event( array $record, int $post_id ): void {
+		if (
+			! empty( $record['_tribe_default_ticket_provider'] )
+		    && $record['_tribe_default_ticket_provider'] == "TEC\Tickets\Commerce\Module"
+		) {
+			if ( $this->fix_ticket_provider( $post_id ) ) {
+				$this->add_to_log( "Ticket provider successfully updated." );
+			} else {
+				$this->add_to_log( "Ticket provider update failed.");
+			}
+		}
+	}
+
+	/**
+	 * Update post_name (new id) and post_parent (new order id) for Tickets Commerce Attendees.
+	 *
+	 * @param array $record  The post data.
+	 * @param int   $post_id The new post ID.
+	 *
+	 * @return void
+	 */
+	public function maybe_update_post_data_for_attendee( array $record, int $post_id ): void {
+		$stop = false;
+
+		// If there is no record for the parent in the source data, then stop.
+		if ( ! isset( $record['parent'] ) ) {
+			$this->add_to_log( 'Post parent missing from import data...' );
+			$stop = true;
+		} else {
+			// Hash the old linked post type ID (tec_tc_order).
+			$this->meta_value = $this->hashit( $record['parent'] );
+			$this->meta_key   = "_tec_tc_order_export_hash";
```

Comment:
Question: this is only dealing with Commerce tickets, right?

Reply:
@lucatume Yes, for now only Tickets Commerce.